Client/Frontend/Intro/IntroScreenWelcomeViewV2.swift

```diff
@@ -0,0 +1,281 @@
+            closeButton.tintColor = .secondaryLabel
+        } else {
+            closeButton.tintColor = .black
+        }
+        return closeButton
+    }()
+    private lazy var nextButton: UIButton = {
+        let button = UIButton()
+        button.setTitle(Strings.IntroNextButtonTitle, for: .normal)
+        button.titleLabel?.font = UIFont.systemFont(ofSize: 18, weight: .semibold)
+        button.setTitleColor(UIColor.Photon.Blue50, for: .normal)
+        button.titleLabel?.textAlignment = .center
+        return button
+    }()
+    // Welcome card items share same type of label hence combining them into a
+    // struct so we can reuse it
+    private struct WelcomeUICardItem {
+        var title:String
+        var description:String
+        var titleColour: UIColor
+        var descriptionColour: UIColor
+        lazy var titleLabel: UILabel = {
+            let label = UILabel()
+            label.text = title
+            label.textColor = titleColour
+            label.font = UIFont.systemFont(ofSize: 19, weight: .semibold)
+            label.textAlignment = .left
+            label.numberOfLines = 1
+            label.adjustsFontSizeToFitWidth = true
+            return label
+        }()
+        lazy var descriptionLabel: UILabel = {
+            let label = UILabel()
+            label.text = description
+            label.textColor = descriptionColour
+            label.font = UIFont.systemFont(ofSize: 19, weight: .regular)
+            label.textAlignment = .left
+            label.numberOfLines = 2
+            label.adjustsFontSizeToFitWidth = true
+            return label
+        }()
+    }
+    private lazy var welcomeCardItems: [WelcomeUICardItem] = {
+        var cardItems = [WelcomeUICardItem]()
+        // Automatic Privacy
+        let automaticPrivacy = WelcomeUICardItem(title: Strings.CardTitleAutomaticPrivacy, description: Strings.CardDescriptionAutomaticPrivacy, titleColour: fxTextThemeColour, descriptionColour: fxTextThemeColour)
+        cardItems.append(automaticPrivacy)
+        // Fast Search
+        let fastSearch = WelcomeUICardItem(title: Strings.CardTitleFastSearch, description: Strings.CardDescriptionFastSearch, titleColour: fxTextThemeColour, descriptionColour: fxTextThemeColour)
+        cardItems.append(fastSearch)
+        // Safe Sync
+        let safeSync = WelcomeUICardItem(title: Strings.CardTitleSafeSync, description: Strings.CardDescriptionSafeSync, titleColour: fxTextThemeColour, descriptionColour: fxTextThemeColour)
+        cardItems.append(safeSync)
+        return cardItems
+    }()
+    // See above for explanation of each of these views
+    private var topView = UIView()
+    private var automaticPrivacyView = UIView()
+    private var fastSearchView = UIView()
+    private var safeSyncView = UIView()
+    private var itemStackView = UIStackView()
+    private var combinedView = UIView()
+    // Screen constants
+    private let screenHeight = UIScreen.main.bounds.size.height
```

Comment:
this is variable on orientation, is there a common utility function that does this safely?

Reply:
Usually we would do something like view.frame.height, but problem is because initially its view without a frame that gives 0. I am open to suggestions on how to get the height

Reply:
found it, use `DeviceInfo.screenSizeOrientationIndependent()`